generator/discovery/update_discovery_doc.sh

```diff
@@ -19,6 +19,26 @@ set -euo pipefail
 source "$(dirname "$0")/../../ci/lib/init.sh"
 source module ci/lib/io.sh
 
+function print_service_textproto() {
+  service_proto_path=$(echo "${1}" | sed -En 's/protos\/(google\/cloud\/compute\/.*\/v[[:digit:]]\/.*\.proto)/\1/p')
```

Comment:
I believe you can do that with some magic shell expansion, but whether it is more readable or less readable I am not sure.  Maybe ask @devbww who has forgotten more about shell than I would ever learn.

Reply:
It seems to me that, with the additional knowledge that the final `.*` could be replaced by `[^/]*`, that this would suffice:
```
  service_proto_path="${1#protos/}"
  product_path="${service_proto_path%/*\.proto}"
```